lib/framework/MQTTSettingsService.cpp

```diff
@@ -0,0 +1,137 @@
+#include <MQTTSettingsService.h>
+
+MQTTSettingsService::MQTTSettingsService(AsyncWebServer* server, FS* fs, SecurityManager* securityManager) :
+    _settingsEndpoint(MQTTSettings::serialize,
+                      MQTTSettings::deserialize,
+                      this,
+                      server,
+                      MQTT_SETTINGS_SERVICE_PATH,
+                      securityManager),
+    _settingsPersistence(MQTTSettings::serialize, MQTTSettings::deserialize, this, fs, MQTT_SETTINGS_FILE) {
+#ifdef ESP32
+  WiFi.onEvent(
+      std::bind(&MQTTSettingsService::onStationModeDisconnected, this, std::placeholders::_1, std::placeholders::_2),
+      WiFiEvent_t::SYSTEM_EVENT_STA_DISCONNECTED);
+  WiFi.onEvent(std::bind(&MQTTSettingsService::onStationModeGotIP, this, std::placeholders::_1, std::placeholders::_2),
+               WiFiEvent_t::SYSTEM_EVENT_STA_GOT_IP);
+#elif defined(ESP8266)
+  _onStationModeDisconnectedHandler = WiFi.onStationModeDisconnected(
+      std::bind(&MQTTSettingsService::onStationModeDisconnected, this, std::placeholders::_1));
+  _onStationModeGotIPHandler =
+      WiFi.onStationModeGotIP(std::bind(&MQTTSettingsService::onStationModeGotIP, this, std::placeholders::_1));
+#endif
+  _mqttClient.onConnect(std::bind(&MQTTSettingsService::onMqttConnect, this, std::placeholders::_1));
+  _mqttClient.onDisconnect(std::bind(&MQTTSettingsService::onMqttDisconnect, this, std::placeholders::_1));
+  addUpdateHandler([&](String originId) { onConfigUpdated(); }, false);
+}
+
+MQTTSettingsService::~MQTTSettingsService() {
+}
+
+void MQTTSettingsService::begin() {
+  _settingsPersistence.readFromFS();
+}
+
+void MQTTSettingsService::loop() {
+  if (_reconfigureMqtt || (_disconnectedAt && (unsigned long)(millis() - _disconnectedAt) >= MQTT_RECONNECTION_DELAY)) {
+    // reconfigure MQTT client
+    configureMQTT();
+
+    // clear the reconnection flags
+    _reconfigureMqtt = false;
+    _disconnectedAt = 0;
+  }
+}
+
+bool MQTTSettingsService::isEnabled() {
+  return _settings.enabled;
+}
+
+bool MQTTSettingsService::isConnected() {
+  return _connected;
```

Comment:
```suggestion
  return _mqttClient.connected();
```

Can we use connected method of AsyncMqttClient ?

Reply:
Yes, I can't think of a legitimate reason for tracking the connection status in two places. Now delegating to the client's connected() function instead.